src/dropdown/dropdown.service.ts

```diff
@@ -111,22 +111,24 @@ export class DropdownService implements OnDestroy {
 
 		const boxMenu = menuRef.querySelector(".bx--list-box__menu");
 
-		// If the parentRef and boxMenu are in a different left position relative to the
-		// window, the the boxMenu position has already been flipped and a check needs to be done
-		// to see if it needs to stay flipped.
-		if (parentRef.getBoundingClientRect().left !== boxMenu.getBoundingClientRect().left) {
-			// The getBoundingClientRect().right of the boxMenu if it were hypothetically flipped
-			// back into the original position before the flip.
-			const testBoxMenuRightEdgePos =
-				parentRef.getBoundingClientRect().left - boxMenu.getBoundingClientRect().left + boxMenu.getBoundingClientRect().right;
-
-			if (testBoxMenuRightEdgePos > (window.innerWidth || document.documentElement.clientWidth)) {
+		if (boxMenu) {
+			// If the parentRef and boxMenu are in a different left position relative to the
+			// window, the the boxMenu position has already been flipped and a check needs to be done
+			// to see if it needs to stay flipped.
+			if (parentRef.getBoundingClientRect().left !== boxMenu.getBoundingClientRect().left) {
+				// The getBoundingClientRect().right of the boxMenu if it were hypothetically flipped
+				// back into the original position before the flip.
+				const testBoxMenuRightEdgePos =
+					parentRef.getBoundingClientRect().left - boxMenu.getBoundingClientRect().left + boxMenu.getBoundingClientRect().right;
+
+				if (testBoxMenuRightEdgePos > (window.innerWidth || document.documentElement.clientWidth)) {
+					leftOffset = parentRef.offsetWidth - boxMenu.offsetWidth;
+				}
+			// If it has not already been flipped, check if it is necessary to flip, ie. if the
+			// boxMenu is outside of the right viewPort.
+			} else if (boxMenu.getBoundingClientRect().right > (window.innerWidth || document.documentElement.clientWidth)) {
```

Comment:
Not sure if there are Gatsby-like frameworks that build angular in an SSR environment. Would the direct use of `window` reference cause an issue there?

Reply:
I'm not familiar with what you describe, but i guess it would provide a global window object to mock browser behaviour. Anyway this line was already there...my change in this file is just putting everything using `boxMenu` under an `if` checking it to be defined.

cc @Donisius who introduced the line you commented...he might better answer your question

Reply:
In the case of Gatsby for instance, it wouldn't mock `window` by default, so you'd write something like `const isBrowser = typeof window !== "undefined"`

Reply:
> Not sure if there are Gatsby-like frameworks that build angular in an SSR environment. Would the direct use of `window` reference cause an issue there?

Yeah, Angular Universal is Angular's SSR framework ... We have a number of areas with similar issues, eventually we'll have a service to mock out `window` which will let us solve all of these issues at once ... I just haven't gotten around to it yet 😛